### ISSUE TITLE
Rendering issue with fullscreened applications

### ISSUE THREAD

**Issue by sodface** [project-repo/cagebreak#71]
Hi, running the latest development branch so I'm not sure if this occurs on previous versions, but I noticed when going fullscreen with mpv (key f or double click) or firefox (F11) the window rendering gets wonky.

This can be fixed by just re-selecting the workspace.

This does not occur with a single monitor. On my dual monitor desktop, it's as if the application thinks it needs to span the combined resolution of both monitors when first entering fullscreen mode. After re-selecting the workspace it redraws correctly.

With firefox for example, pressing F11 takes you from screenshot 1 to screenshot 2:

![20240108_07h53m02s_grim](https://github.com/project-repo/cagebreak/assets/46695384/37cb5f63-fad7-422a-a0b1-6ce757ec5b1d)

![20240108_07h53m11s_grim](https://github.com/project-repo/cagebreak/assets/46695384/7721f77d-e688-415d-b832-ba8b5242f996)

mpv behaves similarly:

![20240108_07h47m56s_grim](https://github.com/project-repo/cagebreak/assets/46695384/49ae8b5d-44be-4c03-a394-c9e534ead17b)

![20240108_07h48m05s_grim](https://github.com/project-repo/cagebreak/assets/46695384/5923a62f-978b-4e2a-92ed-a89f9008c111)

**Comment by project-repo** [comment]
Thanks for the bug report! I think we were able to locate the issue,
could you try out the latest commits on the development branch?
Cheers,
project-repo

**Comment by sodface** [comment]
That does seem to fix the fullscreen issue but it causes by desktop layout to be left(2) <-- right(1) even with config:

```
output HDMI-A-1 prio 1
output HDMI-A-1 permanent
output HDMI-A-2 permanent
```

I tried commenting out the config too but the result was the same.

**Comment by sodface** [comment]
If it helps, I went back a few commits in the development branch and skipped most of the ones that I thought weren't relevant to functionality.

This commit affects my layout as noted in the above comment:
https://github.com/project-repo/cagebreak/commit/4b4f6d42745971ba9869bc4ce60dd878cefb0ffb

This commit is working ok:
https://github.com/project-repo/cagebreak/commit/3da019876823ff670d420391a5b692e7c662c3b9

**Comment by sodface** [comment]
Sorry, correction:

Launching this commit from tty just crashes back to tty:
https://github.com/project-repo/cagebreak/commit/4b4f6d42745971ba9869bc4ce60dd878cefb0ffb

This commit launches ok but the dual monitor layout is incorrect:
https://github.com/project-repo/cagebreak/commit/99b643a65b74d7d8f3d2d5bf5ba490ba316fa93e

Layout is correct with this commit:
https://github.com/project-repo/cagebreak/commit/3da019876823ff670d420391a5b692e7c662c3b9

**Comment by project-repo** [comment]
Hi sodface

Sorry for not getting back sooner... Thanks for the bisect though, could
you give the latest commit on development a spin? Does that change
anything?

Cheers
project-repo

**Comment by sodface** [comment]
Quick test since I'm heading out but it looks good, thanks!

**Comment by sodface** [comment]
This can be closed too correct? 2.3.0 works well.

**Comment by project-repo** [comment]
Fixed with 2.3.0

Closing.

cheers
project-repo
